File: Cpp/151_Reverse_Words_in_a_String.cpp
class Solution {
public:
    string reverseWords(string s) {
        string result = "";
        stack<string> word_stack;
        string word = "";
        
        for(auto& ch : s){
            if(ch == ' ') {
                if(word.empty()) continue;
                word_stack.push(word);
                word = "";
            } else {
                word += ch;
            }
        }
        
        if(!word.empty()) word_stack.push(word);
        
        if(word_stack.empty()) return "";
        
        while(!word_stack.empty()){
            result += word_stack.top() + ' ';
            word_stack.pop();
        }
        result.pop_back();
        
        return result;
    }
};

File: Kotlin/151_Reverse_Words_in_a_String.kt
class Solution {
    fun reverseWords(s: String): String {
        val result = StringBuilder()
        val st = Stack<String>()
        val word = StringBuilder()
        
        for(ch in s){
            if(ch.isWhitespace()){
                if(word.isNotBlank()){
                    st.push(word.toString())
                }
                word.clear()
            }else{
                word.append(ch)
            }
        }
        st.push(word.toString())
        
        while(st.isNotEmpty()){
            result.append(st.pop())
            result.append(" ")
        }
        
        return result.trim().toString()
    }
}